### ISSUE TITLE
syft seems unable to catalog .jar files on Windows

### ISSUE THREAD

**Issue by westonsteimel** [anchore/syft#683]
**What happened**:

Using syft on windows (release from [here](https://github.com/anchore/syft/releases/download/v0.32.1/syft_0.32.1_windows_amd64.zip)), it was unable to identify any `java-archive` packages from a single .jar file in a directory.  The same file does get correctly cataloged when running syft from Linux.

**What you expected to happen**:

syft to correctly identify package dependencies from a .jar on Windows

**How to reproduce it (as minimally and precisely as possible)**:

Any valid .jar file should do, but I used [AthenaJDBC42_2.0.5.jar](https://s3.amazonaws.com/athena-downloads/drivers/JDBC/SimbaAthenaJDBC_2.0.5/AthenaJDBC42_2.0.5.jar).  Place it in a directory on Windows.  

```powershell
./syft.exe -vvv dir:~/test
```
<details>
  <summary>Output</summary>

  ```powershell
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m application config:
  output: table
  file: ""
  quiet: false
  check-for-app-update: true
  anchore:
    host: ""
    path: ""
    dockerfile: ""
    overwrite-existing-image: false
    import-timeout: 30
  dev:
    profile-cpu: false
    profile-mem: false
  log:
    structured: false
    level: debug
    file: ""
  package:
    cataloger:
      enabled: true
      scope: Squashed
  file-metadata:
    cataloger:
      enabled: false
      scope: Squashed
    digests:
    - sha256
  file-classification:
    cataloger:
      enabled: false
      scope: Squashed
  file-contents:
    cataloger:
      enabled: false
      scope: Squashed
    skip-files-above-size: 1048576
    globs: []
    cataloger:
      enabled: false
      scope: AllLayers
    additional-patterns: {}
    exclude-pattern-names: []
    reveal-values: false
    skip-files-above-size: 1048576
  registry:
    insecure-skip-tls-verify: false
    insecure-use-http: false
    auth: []

  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m no new syft update available
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m indexing filesystem path="C:\\Users\\weston\\test"
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m path not found: /usr/lib/os-release
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m path not found: /bin/busybox
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m path not found: /etc/system-release-cpe
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m path not found: /etc/redhat-release
  ←[0;90m[0000]←[0m ←[0;32m INFO←[0m could not identify distro
  ←[0;90m[0000]←[0m ←[0;32m INFO←[0m cataloging directory
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m cataloging with "ruby-gemfile-cataloger"
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m discovered 0 packages
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m cataloging with "python-index-cataloger"
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m discovered 0 packages
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m discovered 0 packages
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m cataloging with "php-composer-lock-cataloger"
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m discovered 0 packages
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m cataloging with "javascript-lock-cataloger"
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m discovered 0 packages
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m cataloging with "dpkgdb-cataloger"
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m discovered 0 packages
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m cataloging with "rpmdb-cataloger"
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m discovered 0 packages
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m cataloging with "java-cataloger"
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m cataloging with "apkdb-cataloger"
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m discovered 0 packages
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m cataloging with "go-module-binary-cataloger"
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m discovered 0 packages
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m cataloging with "go-mod-file-cataloger"
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m discovered 0 packages
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m cataloging with "rust-cataloger"
  ←[0;90m[0000]←[0m ←[0;34mDEBUG←[0m discovered 0 packages
  No packages discovered
  ```

</details>

And on Linux

```bash
./syft -vvv dir:~/test
```

<details>
  <summary>Output</summary>

  ```bash
  [0000] DEBUG application config:
  output: table
  file: ""
  quiet: false
  check-for-app-update: true
  anchore:
    host: ""
    path: ""
    dockerfile: ""
    overwrite-existing-image: false
    import-timeout: 30
  dev:
    profile-cpu: false
    profile-mem: false
  log:
    structured: false
    level: debug
    file: ""
  package:
    cataloger:
      enabled: true
      scope: Squashed
  file-metadata:
    cataloger:
      enabled: false
      scope: Squashed
    digests:
    - sha256
  file-classification:
    cataloger:
      enabled: false
      scope: Squashed
  file-contents:
    cataloger:
      enabled: false
      scope: Squashed
    skip-files-above-size: 1048576
    globs: []
  secrets:
    cataloger:
      enabled: false
      scope: AllLayers
    additional-patterns: {}
    exclude-pattern-names: []
    reveal-values: false
    skip-files-above-size: 1048576
  registry:
    insecure-skip-tls-verify: false
    insecure-use-http: false
    auth: []
  
  [0000] DEBUG no new syft update available
  [0000] DEBUG indexing filesystem path="/home/weston/test"
  [0000] DEBUG path not found: /etc/os-release
  [0000] DEBUG path not found: /usr/lib/os-release
  [0000] DEBUG path not found: /bin/busybox
  [0000] DEBUG path not found: /etc/system-release-cpe
  [0000] DEBUG path not found: /etc/redhat-release
  [0000]  INFO could not identify distro
  [0000]  INFO cataloging directory
  [0000] DEBUG cataloging with "ruby-gemfile-cataloger"
  [0000] DEBUG discovered 0 packages
  [0000] DEBUG cataloging with "python-index-cataloger"
  [0000] DEBUG discovered 0 packages
  [0000] DEBUG cataloging with "python-package-cataloger"
  [0000] DEBUG discovered 0 packages
  [0000] DEBUG cataloging with "php-composer-lock-cataloger"
  [0000] DEBUG discovered 0 packages
  [0000] DEBUG cataloging with "javascript-lock-cataloger"
  [0000] DEBUG discovered 0 packages
  [0000] DEBUG cataloging with "dpkgdb-cataloger"
  [0000] DEBUG discovered 0 packages
  [0000] DEBUG cataloging with "rpmdb-cataloger"
  [0000] DEBUG discovered 0 packages
  [0000] DEBUG cataloging with "java-cataloger"
  [0000] DEBUG discovered 31 packages
  [0000] DEBUG cataloging with "apkdb-cataloger"
  [0000] DEBUG discovered 0 packages
  [0000] DEBUG cataloging with "go-module-binary-cataloger"
  [0000] DEBUG discovered 0 packages
  [0000] DEBUG cataloging with "go-mod-file-cataloger"
  [0000] DEBUG discovered 0 packages
  [0000] DEBUG cataloging with "rust-cataloger"
  [0000] DEBUG discovered 0 packages
  NAME                      VERSION      TYPE
  AthenaJDBC42              2.0.5.1006   java-archive
  AthenaJDBC42_2.0.5                     java-archive
  RedshiftJDBC              1.2.16.1027  java-archive
  RedshiftJDBC42            1.2.16.1027  java-archive
  RedshiftJDBC42-no-awssdk  1.2.20.1043  java-archive
  aws-java-sdk-athena       1.11.335     java-archive
  aws-java-sdk-core         1.11.118     java-archive
  aws-java-sdk-core         1.11.335     java-archive
  aws-java-sdk-glue         1.11.335     java-archive
  aws-java-sdk-redshift     1.11.118     java-archive
  aws-java-sdk-sts          1.11.118     java-archive
  aws-java-sdk-sts          1.11.335     java-archive
  commons-codec             1.10         java-archive
  commons-codec             1.9          java-archive
  commons-csv               1.4          java-archive
  commons-logging           1.1.3        java-archive
  commons-logging           1.2          java-archive
  httpclient                4.5.2        java-archive
  httpclient                4.5.3        java-archive
  httpcore                  4.4.4        java-archive
  httpcore                  4.4.6        java-archive
  jackson-annotations       2.6.0        java-archive
  jackson-annotations       2.9.5        java-archive
  jackson-core              2.6.6        java-archive
  jackson-core              2.9.5        java-archive
  jackson-databind          2.6.6        java-archive
  jackson-databind          2.9.5        java-archive
  jackson-dataformat-cbor   2.6.6        java-archive
  joda-time                 2.8.1        java-archive
  joda-time                 2.9.9        java-archive
  log4j                     1.2.17       java-archive

  ```
</details>

**Anything else we need to know?**:

I tried as far back as version 0.28.0 and none of them seemed to find java-archives on Windows.  I did try a go binary and it seemed to correctly work for that.  I haven't tried any other package types yet.

**Environment**:
- Output of `syft version`: 
```powershell
Application:   syft
Version:       0.32.1
BuildDate:     2021-12-14T02:10:55Z
GitCommit:     d3804d1a827c9b1fa9867208de1167a9e62c6e99
GitTreeState:  clean
Platform:      windows/amd64
GoVersion:     go1.16.11
Compiler:      gc
```
- OS (e.g: `cat /etc/os-release` or similar): Windows

**Comment by Crysinyan** [comment]
Can confirm for multiple package types.

Windows:
<details>
<summary>Output</summary>

```
[0000] DEBUG no new syft update available
[0000] DEBUG indexing filesystem path="C:\\Temp\\test"
[0001] DEBUG path not found: /etc/os-release
[0001] DEBUG path not found: /usr/lib/os-release
[0001] DEBUG path not found: /bin/busybox
[0001] DEBUG path not found: /etc/system-release-cpe
[0001] DEBUG path not found: /etc/redhat-release
[0001]  INFO could not identify distro
[0001]  INFO cataloging directory
[0001] DEBUG cataloging with "ruby-gemfile-cataloger"
[0001] DEBUG discovered 0 packages
[0001] DEBUG cataloging with "python-index-cataloger"
[0001] DEBUG discovered 0 packages
[0001] DEBUG cataloging with "python-package-cataloger"
[0001] DEBUG discovered 0 packages
[0001] DEBUG cataloging with "php-composer-lock-cataloger"
[0001] DEBUG discovered 0 packages
[0001] DEBUG cataloging with "javascript-lock-cataloger"
[0001] DEBUG discovered 0 packages
[0001] DEBUG cataloging with "dpkgdb-cataloger"
[0001] DEBUG discovered 0 packages
[0001] DEBUG cataloging with "rpmdb-cataloger"
[0001] DEBUG discovered 0 packages
[0001] DEBUG cataloging with "java-cataloger"
[0002] DEBUG discovered 0 packages
[0002] DEBUG cataloging with "apkdb-cataloger"
[0002] DEBUG discovered 0 packages
[0002] DEBUG cataloging with "go-module-binary-cataloger"
[0002] DEBUG discovered 91 packages
[0002] DEBUG cataloging with "go-mod-file-cataloger"
[0002] DEBUG discovered 0 packages
[0002] DEBUG cataloging with "rust-cataloger"
[0002] DEBUG discovered 0 packages
```
</details>

Linux in WSL:
<details>
<summary>Output</summary>

```
[0000] DEBUG no new syft update available
[0000] DEBUG indexing filesystem path="/mnt/c/Temp/test"
[0057] DEBUG path not found: /etc/os-release
[0057] DEBUG path not found: /usr/lib/os-release
[0057] DEBUG path not found: /bin/busybox
[0057] DEBUG path not found: /etc/system-release-cpe
[0057] DEBUG path not found: /etc/redhat-release
[0057]  INFO could not identify distro
[0057]  INFO cataloging directory
[0057] DEBUG cataloging with "ruby-gemfile-cataloger"
[0057] DEBUG discovered 102 packages
[0057] DEBUG cataloging with "python-index-cataloger"
[0058] DEBUG discovered 22 packages
[0058] DEBUG cataloging with "python-package-cataloger"
[0058] DEBUG discovered 8 packages
[0058] DEBUG cataloging with "php-composer-lock-cataloger"
[0059] DEBUG discovered 4 packages
[0059] DEBUG cataloging with "javascript-lock-cataloger"
[0059] DEBUG discovered 32 packages
[0059] DEBUG cataloging with "dpkgdb-cataloger"
[0059] DEBUG cataloging with "rpmdb-cataloger"
[0059] DEBUG discovered 1 packages
[0059] DEBUG cataloging with "java-cataloger"
[0068] DEBUG discovered 750 packages
[0068] DEBUG cataloging with "apkdb-cataloger"
[0068] DEBUG discovered 2 packages
[0068] DEBUG cataloging with "go-module-binary-cataloger"
[0071] DEBUG discovered 91 packages
[0072] DEBUG cataloging with "go-mod-file-cataloger"
[0072] DEBUG discovered 54 packages
[0072] DEBUG cataloging with "rust-cataloger"
[0072] DEBUG discovered 13 packages
No packages discovered
1 error occurred:
        * unable to catalog dpkg package=Go/syft-old/test/integration/test-fixtures/image-pkg-coverage/pkgs/var/lib/dpkg/status: cannot p'rse field from line: '
```
</details>


On a different note, seems the package the linux verison is trying to catalog is a folder.


Version:
```
Application:   syft
Version:       0.32.1
BuildDate:     2021-12-14T02:10:55Z
GitCommit:     d3804d1a827c9b1fa9867208de1167a9e62c6e99
GitTreeState:  clean
Platform:      windows/amd64
GoVersion:     go1.16.11
Compiler:      gc
```

**Comment by luhring** [comment]
Thanks for the info @westonsteimel and @Crysinyan — we're looking into this now.

**Comment by spiffcs** [comment]
Closed by #699
